tests/MediaTest.php

```diff
@@ -0,0 +1,340 @@
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => $this->user->id,
+            'is_public' => true,
+            'link' => '/storage/file.png',
+        ]);
+    }
+
+    public function testCreateCheckUrls(): void
+    {
+        $this->mockGenerateFilename();
+
+        $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $this->assertEquals(1, Media::where('link', 'like', '/%')->count());
+    }
+
+    public function testCreateCheckResponse(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'link' => '/storage/file.png',
+        ]);
+
+        Storage::disk('local')->assertExists($this->getFilePathFromUrl('file.png'));
+
+        $this->clearUploadedFilesFolder();
+    }
+
+    public function testCreateNoAuth(): void
+    {
+        $response = $this->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertUnauthorized();
+    }
+
+    public function testBulkCreate(): void
+    {
+        /*
+        $this->mockGenerateFilename(2);
+        */
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->admin)->json('post', '/media/bulk', [
+            'media' => [
+                [
+                    'file' => $this->file,
+                    'meta' => ['test1'],
+                ],
+                /*
+                [
+                    'file' => $this->file,
+                    'meta' => ['test2'],
+                ],
+                */
+            ],
+        ]);
+
+        $response->assertOk();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => $this->admin->id,
+            'meta' => "[\"test1\"]",
+            'is_public' => false,
+        ]);
+
+        /*
+        $this->assertDatabaseHas('media', [
+            'id' => 7,
+            'name' => 'file.png',
+            'owner_id' => $this->admin->id,
+            'meta' => "[\"test2\"]",
+            'is_public' => false,
+        ]);
+        */
+    }
+
+    public function testDelete(): void
+    {
+        $response = $this->actingAs($this->admin)->json('delete', '/media/1');
+
+        $response->assertNoContent();
+
+        $this->assertSoftDeleted('media', [
+            'id' => 1,
+        ]);
+    }
+
+    public function testDeleteNotExists(): void
+    {
+        $response = $this->actingAs($this->admin)->json('delete', '/media/0');
+
+        $response->assertNotFound();
+    }
+
+    public function testDeleteNoPermission(): void
+    {
+        $response = $this->actingAs($this->user)->json('delete', '/media/1');
+
+        $response->assertForbidden();
+
+        $this->assertDatabaseHas('media', [
```

Comment:
please use `assertNoChanges` from the LaravelHelpers package

Reply:
As I already wrote, one problem arose during the development of the package.

For tests to work, they must inherit from Orchestra\Testbench\TestCase.
To take full advantage of LaravelHelpers, tests must inherit from RonasIT\Support\Tests\TestCase.

I created a new class, took some code from the LaravelHelpers package and inherited it from Orchestra\Testbench\TestCase.

To use assertNoChanges, I'll have to copy quite a bit from LaravelHelpers. Is it really worth it?

Reply:
this method is related to the model test state class https://github.com/RonasIT/laravel-helpers/blob/master/src/Tests/ModelTestState.php so you can use it without extending base test case

Reply:
Done.